### PR TITLE
(#3584) - overhaul api docs

### PR DESCRIPTION
**Pull request by nolanlawson**
Based on discussion in https://github.com/pouchdb/pouchdb/pull/3577.

Rebased on #3583 for my own sanity.

### CODE REVIEW

**Comment by nolanlawson** [comment]
Done! Man, I am so happy with how this turned out. This is what I always wanted the docs to look like: nice, clean promise examples, with the ugly callback examples hidden behind a little switcher. :)

I also took this opportunity to add a bunch more documentation where it was missing. For instance:

* attachments, with examples of using base64 vs blobs/buffers
* more about query, with a little advertisement for pouchdb-find
* cleaned up the events description for changes/replicate
* describe how to use the promise/callback API for changes/replicate
* more docs for remove()

So basically it's just better overall, and @nickcolley did an excellent job with the promise/callback switcher - it even saves your choice, and switches all examples across the entire page. In the future, maybe we could use it for ES7 async functions as well? :guitar:

**Comment by nolanlawson** [comment]
Also, for stuff that's deprecated (`'continuous'`, `'latest'`, `PouchDB.destroy`), I just removed references to them. The only thing I kept was `'uptodate'`, because that was something we deprecated pretty recently.

**Comment by nickcolley** [comment]
Think we need to address the changing of height when switching, if you're at the bottom of the page it can jump out of view.

**Comment by nickcolley** [comment]
Feel free to squash the border-radius commit into the previous one, I just didn't want to force push a shared branch.

**Comment by nickcolley** [comment]
Looks like somehow the manifest got deleted?

Also, @nolanlawson try renaming the api partials to .md I think they should work now.

**Comment by nolanlawson** [comment]
OK, I added `/static/js/code.min.js ` and squashed as much as I could. Looks great now without the rave party jumpiness! :)

**Comment by daleharvey** [comment]
Hrm, getting 

Error:   Liquid Exception: Included file '_includes/alert_start.html' not found in _posts/2015-02-28-efficiently-managing-ui-state-in-pouchdb.md

**Comment by daleharvey** [comment]
Fixed it, carry on

**Comment by daleharvey** [comment]
https://github.com/pouchdb/pouchdb/commit/2f6280d3156428f0ed751e743e0ec4b3596ca7e5
https://github.com/pouchdb/pouchdb/commit/f635552e2869e88cc0e5bb4de19d147e77dca2dc

Awesome job yall, this is great
